### ISSUE TITLE
[BUG] MDCA - Logic app success on 401

### ISSUE THREAD

**Issue by briandelmsft** [briandelmsft/SentinelAutomationModules#223]
**Describe the bug**
The logic app shows successful run even when a 401 is encountered
Randomly see 401s with 

```
{
  "detail": "Invalid JSON Web Key Set"
}
```

**Module Name**
MCAS/MDCA

**Comment by piaudonn** [comment]
🙃 This might be a timing issue with recently created user. Need to confirm with support or engineering...

**Comment by briandelmsft** [comment]
Yeah i figured, might not be much we can do but we need to make sure the logic app returns a fail instead of a success in that case with a terminate action after the response.

**Comment by piaudonn** [comment]
The problem seems to be related to the use of the generic URL `portal.cloudappsecurity.com` in the API call.
DNS round robin makes the module call the wrong backend once in a while.
Note that this URL was used to avoid having the operator installing the solution to go grab the MCAS API URL before continuing as it seems there is no programatic way to retrieve or calculate this URL.

I suggest this workaround: https://github.com/briandelmsft/SentinelAutomationModules/commit/0d4778533f7d29d50d3a0fec206ca0c54e9b48cc

Which basically is trying the 5 known (as of today) of Tenant Region for Microsoft Defender for Cloud Apps (according to the [privacy documentation](https://docs.microsoft.com/en-us/defender-cloud-apps/cas-compliance-trust#data-location) and the Content-Security-Policy header returned by the API itself: `*.eu.portal.cloudappsecurity.com *.us.portal.cloudappsecurity.com *.us2.portal.cloudappsecurity.com *.eu2.portal.cloudappsecurity.com *.us3.portal.cloudappsecurity.com`).
![image](https://user-images.githubusercontent.com/22434561/153328692-d9c6e443-782d-4acb-a935-0df068c41fa9.png)

That's quick (the checks are made in parallel) but not super elegant. As soon as we found the right URL we use it for the actual lookup. This doesn't require any permissions change.

Not elegant, but doesn't affect the installation.

In another hand, we can also just ask for the API URL at install and update time.

Thoughts?

**Comment by piaudonn** [comment]
@briandelmsft following our conversation:

Moved the Initialize APIURL step on the top:
![image](https://user-images.githubusercontent.com/22434561/153332415-f18a403b-8d3e-4002-91fa-6ccd807df841.png)

Then we enter in the "best effort loop" only if the user hasn't set any:
![image](https://user-images.githubusercontent.com/22434561/153332503-037dda96-c52f-427c-9555-e50183aea007.png)

And I update the documentation to reflect this.
